lib/pka_dev.c

```diff
@@ -1726,11 +1727,22 @@ bool pka_dev_has_avail_ring(pka_ring_info_t *ring_info,
         }
 
         // Open the ring corresponding to the given ring identifier.
-        if (pka_dev_open_ring(ring_info))
+        ret = pka_dev_open_ring(ring_info);
+        if (ret == -EBUSY)
+        {
+            PKA_DEBUG(PKA_USER, "Ring %d is busy\n", ring_info->ring_id);

```

Comment:
This is the "device" interface (i.e., pka_dev), I think we should use PKA_DEV rather than PKA_USER